pkg/mcproducer/scripts/replace-kernel-module.sh

```diff
@@ -19,17 +23,42 @@ EOF
     cat "$kmm_config_file_filepath"
 }
 
-echo "before checking image tar file presence"
-if [ -e $kernel_module_image_filepath ]; then
-    echo "Image file $kernel_module_image_filepath found on the local file system, creating kmm config file"
+echo "before checking image presence"
+if [ -n "$(podman images -q $full_kernel_module_image 2> /dev/null)" ]; then
+    echo "Image $full_kernel_module_image found on the local file system, creating kmm config file"
     create_kmm_config
-    echo "running kernel-management worker image"
-    podman run --user=root --privileged -v /lib/modules:/lib/modules -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml -v $kernel_module_image_filepath:$kernel_module_image_filepath $worker_image kmod load --tarball /etc/kmm-worker/config.yaml
+    echo "creating volume"
+    podman volume create $worker_volume_name
+    podman pod create --name $worker_pod_name
+    echo "creating init container"
+    podman create \
+      --pod $worker_pod_name \
+      --init-ctr=always \
+      -v $worker_volume_name:/tmp \
+      $full_kernel_module_image \
+      /bin/sh -c "mkdir -p /tmp/opt/lib/modules && \
+      cp -R /opt/lib/modules/* /tmp/opt/lib/modules"
+    echo "creating worker container"
+    worker_pod_id=$(
+    podman create \
+      --pod $worker_pod_name\
+      --user=root \
+      --privileged \
+      -v $worker_volume_name:/tmp \
+      -v /lib/modules:/lib/modules \
+      -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml \
+      $worker_image \
+      kmod load /etc/kmm-worker/config.yaml)
+    echo "running worker pod"
+    podman pod start $worker_pod_name
     if [ $? -eq 0 ]; then
         echo "OOT kernel module $kernel_module is inserted"
     else
         echo "failed to insert OOT kernel module $kernel_module"
     fi
+    podman wait $worker_pod_id
+    echo "removing volume"
+    podman volume rm --force $worker_volume_name
```

Comment:
Why do we need the `--force` flag?

Reply:
> Why do we need the `--force` flag?

Because even though the pod is finished, the volume is still being used by the containers. the `--force` will also delete them.

Reply:
I think that if we create the containers with `podman create --rm ...` then when the container is done, the container is removed and then we will be able to safely remove the volume.

If it doesn't work, then we can just merge it with the `--force` flag as you did.

WDYT?

Reply:
yes sure, it works